cl2-component-library/src/components/LateralScrollControls/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import { useTheme } from 'styled-components';
+import Box from '../Box';
+import Button from '../Button';
+import testEnv from '../../utils/testUtils/testEnv';
+interface Props {
+  containerRef?: React.RefObject<HTMLDivElement>; // Ref of the container with elements which can be laterally scrolled through
+  scrollBtnDistanceMobile?: number; // Lateral scroll distance on arrow click for mobile
+  scrollBtnDistanceDesktop?: number; // Lateral scroll distance on arrow click for desktop
+  isSmallerThanPhone?: boolean;
+  children: React.ReactNode;
+}
+
+/*
+ * LateralScrollControls:
+ * Wraps a scrollable container with lateral scroll arrow buttons to scroll left and right.
+ */
+const LateralScrollControls = ({
+  containerRef,
+  scrollBtnDistanceMobile = 200,
+  scrollBtnDistanceDesktop = 350,
+  isSmallerThanPhone,
+  children,
+}: Props) => {
+  const theme = useTheme();
+
+  // Used to determine when the scroll buttons should be disabled (E.g. At scroll end, disable the right button)
+  const [atScrollStart, setAtScrollStart] = useState(true);
+  const [atScrollEnd, setAtScrollEnd] = useState(false);
+
+  containerRef?.current?.addEventListener('scroll', () => {
```

Comment:
Event listeners should be added in a `useEffect` so that we can clean them up when the component unmounts.